common/source_model/src/test/scala/weco/catalogue/source_model/sierra/SierraRulesForRequestingTest.scala

```diff
@@ -0,0 +1,331 @@
+    it("if it's part of the Medical Film & Audio Library") {
+      val testCases =
+        Table("locationCode", "mfgmc", "mfinc", "mfwcm", "hmfac", "mfulc")
+
+      forAll(testCases) {
+        assertBlockedWith(
+          _,
+          expectedMessage =
+            "Item cannot be requested online. Please contact Medical Film & Audio Library.   Email: mfac@wellcome.ac.uk. Telephone: +44 (0)20 76118596/97."
+        )
+      }
+    }
+
+    it("if it needs a manual request") {
+      val testCases = Table(
+        "locationCode",
+        "dbiaa",
+        "dcoaa",
+        "dinad",
+        "dinop",
+        "dinsd",
+        "dints",
+        "dpoaa",
+        "dimgs",
+        "dhuaa",
+        "dimgs",
+        "dingo",
+        "dpleg",
+        "dpuih",
+        "gblip",
+        "ofvds")
+
+      forAll(testCases) {
+        assertBlockedWith(
+          _,
+          expectedMessage =
+            "This item cannot be requested online. Please place a manual request.")
+      }
+    }
+
+    it("if it's with Information Services") {
+      val testCases = Table("locationCode", "isvid", "iscdr")
+
+      forAll(testCases) {
+        assertBlockedWith(
+          _,
+          expectedMessage =
+            "Item cannot be requested online. Please ask at Information Service desk, email: infoserv@wellcome.ac.uk or telephone +44 (0)20 7611 8722."
+        )
+      }
+    }
+
+    it("if it's on the open shelves") {
+      val testCases = Table(
+        "locationCode",
+        "isope",
+        "isref",
+        // Note: this is one of the location codes matched by this rule, but it
+        // matches another rule higher up, so it's commented out.
+        // "gblip",
+        "wghib",
+        "wghig",
+        "wghip",
+        "wghir",
+        "wghxb",
+        "wghxg",
+        "wghxp",
+        "wghxr",
+        "wgmem",
+        "wgmxm",
+        "wgpvm",
+        "wgsee",
+        "wgsem",
+        "wgser",
+        "wqrfc",
+        "wqrfd",
+        "wqrfe",
+        "wqrfp",
+        "wqrfr",
+        "wslob",
+        "wslom",
+        "wslor",
+        "wslox",
+        "wsref",
+        "hgslr",
+        "wsrex"
+      )
+
+      forAll(testCases) {
+        assertBlockedWith(
+          _,
+          expectedMessage =
+            "Item is on open shelves.  Check Location and Shelfmark for location details.")
+      }
+    }
+
+    it("if it needs a manual request slip") {
+      val testCases = Table(
+        "locationCode",
+        "ofvn1",
+        "scmwc",
+        "sgmoh",
+        "somet",
+        "somge",
+        "somhe",
+        "somhi",
+        "somja",
+        "sompa",
+        "sompr",
+        "somsy")
+
+      forAll(testCases) {
+        assertBlockedWith(
+          _,
+          expectedMessage =
+            "Please complete a manual request slip.  This item cannot be requested online.")
+      }
+    }
+
+    it("if it's sepep, rm001 or rmdda") {
```

Comment:
What do these mean?

Reply:
Not sure, they're just the codes from the R4R file.

Reply:
Only stands out because the other tests have human parseable reasons.